solr/core/src/java/org/apache/solr/response/transform/ChildDocTransformer.java

```diff
@@ -109,9 +109,14 @@ public void transform(SolrDocument rootDoc, int rootDocId) {
       // Loop each child ID up to the parent (exclusive).
       for (int docId = calcDocIdToIterateFrom(lastChildId, rootDocId); docId < rootDocId; ++docId) {
 
-        // get the path.  (note will default to ANON_CHILD_KEY if not in schema or oddly blank)
+        // get the path.  (note will default to ANON_CHILD_KEY if schema is not nested or empty string if blank)
         String fullDocPath = getPathByDocId(docId - segBaseId, segPathDocValues);
 
+        if(isNestedSchema && !fullDocPath.contains(transformedDocPath)) {
+          // is not a descendant of the transformed doc, return fast.
+          return;
```

Comment:
woah; shouldn't this be "continue"?  We should have a test that would fail on this bug.  All it would take would be an additional child doc that is not underneath the input root/transformed doc but follows it (as provided on input).  Some of the first docs we iterate over here might not descend from rootDocId

Reply:
Yep, you're right.
I'll investigate further to see why a test did not fail because of this.

Reply:
Added another query to [TestChildDocumentHierarchy#testNonRootChildren](https://github.com/apache/lucene-solr/pull/416/files#diff-9fe0ab006f82be5c6a07d5bb6dbc6da0R243).
This test failed before I changed the return to continue(previous commit), and passes using the latest.